### PR TITLE
[Downgrade PHP X.Y] Consistent usage of getPhpVersion() as constant - 1 in tests

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by TomasVotruba** [comment]
I wonder, why is this change needed?

**Comment by samsonasik** [comment]
For consistency, there are already getphpversion() in downgrade rector tests, so make it consistent with the rest

**Comment by samsonasik** [comment]
rebased.

File: rules/downgrade-php72/tests/Rector/FunctionLike/DowngradeReturnObjectTypeDeclarationRector/DowngradeReturnObjectTypeDeclarationRectorTest.php [hunk 1]
```diff
@@ -33,6 +33,6 @@ protected function getRectorClass(): string
 
     protected function getPhpVersion(): int
     {
-        return PhpVersionFeature::OBJECT_TYPE;
+        return PhpVersionFeature::OBJECT_TYPE - 1;
```
Comment (TomasVotruba) [inline rules/downgrade-php72/tests/Rector/FunctionLike/DowngradeReturnObjectTypeDeclarationRector/DowngradeReturnObjectTypeDeclarationRectorTest.php:36]: This is not needed. Mniminal version to parse `object` type is PHP 7.2.

Comment (TomasVotruba) [reply rules/downgrade-php72/tests/Rector/FunctionLike/DowngradeReturnObjectTypeDeclarationRector/DowngradeReturnObjectTypeDeclarationRectorTest.php:36]: The PHP 7.2 version is already requried above:

image

File: rules/downgrade-php71/tests/Rector/TryCatch/DowngradePipeToMultiCatchExceptionRector/DowngradePipeToMultiCatchExceptionRectorTest.php [hunk 2]
```diff
@@ -29,4 +30,9 @@ protected function getRectorClass(): string
     {
         return DowngradePipeToMultiCatchExceptionRector::class;
     }
+
+    protected function getPhpVersion(): int
+    {
+        return PhpVersionFeature::MULTI_EXCEPTION_CATCH - 1;
+    }
```
Comment (TomasVotruba) [inline rules/downgrade-php71/tests/Rector/TryCatch/DowngradePipeToMultiCatchExceptionRector/DowngradePipeToMultiCatchExceptionRectorTest.php:37]: This does not add any new information.

**Comment by TomasVotruba** [comment]
> For consistency, there are already getphpversion() in downgrade rector tests, so make it consistent with the rest

The PHP version is there only on specific purpose, when needed.
On second though, **`getPhpVersion()` might not be needed at all**. The `@require` should handle minimal version, but downgrade should work on any PHP version.

Could you go the other way and remove those methods instead?

File: rules/downgrade-php72/tests/Rector/FunctionLike/DowngradeReturnObjectTypeDeclarationRector/DowngradeReturnObjectTypeDeclarationRectorTest.php (hunk 1 above)
Comment (samsonasik) [reply rules/downgrade-php72/tests/Rector/FunctionLike/DowngradeReturnObjectTypeDeclarationRector/DowngradeReturnObjectTypeDeclarationRectorTest.php]: removed.

File: rules/downgrade-php71/tests/Rector/TryCatch/DowngradePipeToMultiCatchExceptionRector/DowngradePipeToMultiCatchExceptionRectorTest.php (hunk 2 above)
Comment (samsonasik) [reply rules/downgrade-php71/tests/Rector/TryCatch/DowngradePipeToMultiCatchExceptionRector/DowngradePipeToMultiCatchExceptionRectorTest.php:37]: removed.

**Comment by samsonasik** [comment]
I've removed `getPhpVersion()` method in `downgrade-phpXY` rules tests.

**Comment by samsonasik** [comment]
cleaned up unused `PhpVersionFeature`'s constants as well.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by samsonasik** [comment]
rebased.

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Pull request merged**